src/audio/stft_process/stft_process_common.c

```diff
@@ -0,0 +1,304 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation.
+
+#include <sof/audio/component.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/math/auditory.h>
+#include <sof/math/matrix.h>
+#include <sof/math/sqrt.h>
+#include <sof/math/trig.h>
+#include <sof/math/window.h>
+#include <sof/trace/trace.h>
+
+#include "stft_process.h"
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#if STFT_DEBUG
+extern FILE *stft_debug_fft_in_fh;
+extern FILE *stft_debug_fft_out_fh;
+extern FILE *stft_debug_ifft_out_fh;
+
+static void debug_print_to_file_real(FILE *fh, struct icomplex32 *c, int n)
+{
+	for (int i = 0; i < n; i++)
+		fprintf(fh, "%d\n", c[i].real);
+}
+
+static void debug_print_to_file_complex(FILE *fh, struct icomplex32 *c, int n)
+{
+	for (int i = 0; i < n; i++)
+		fprintf(fh, "%d %d\n", c[i].real, c[i].imag);
```

Comment:
`%d+i%d`?

Reply:
The data is not for humans to read. An Octave script reads it with raw ascii numbers load(), so I don't want to include formatting.